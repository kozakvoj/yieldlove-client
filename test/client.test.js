'use strict';

const client = require("../lib/client");

describe("client", () => {
    test("new client should create a new object", () => {
        const yieldloveClient = new client({key: process.env.KEY});

        expect(yieldloveClient.key).toEqual(process.env.KEY);
    });

    test("should throw error if provided without key", () => {
        expect(() => new client({})).toThrowError("You have to provide api key.");
    });

    test("invalid start date should return error", async () => {
        const yieldloveClient = new client({key: process.env.KEY});

        await yieldloveClient.getData("invalid", "2018-10-01")
            .catch(e => {
                console.log(e);
                expect(e).toMatchObject(Error("Start date is invalid."))
            })
    });

    test("invalid end date should return error", async () => {
        const yieldloveClient = new client({key: process.env.KEY});

        await yieldloveClient.getData("2018-10-01", "invalid")
            .catch(e => {
                console.log(e);
                expect(e).toMatchObject(Error("End date is invalid."))
            })
    });

    test("should return filled data", async () => {
        const yieldloveClient = new client({key: process.env.KEY});

        const res = await yieldloveClient.getData("2018-10-01", "2018-10-01");

        expect(res).toMatchSnapshot();
    });

    test("should return empty data", async () => {
        const yieldloveClient = new client({key: process.env.KEY});

        const res = await yieldloveClient.getData("2000-10-01", "2000-10-01");

        expect(res).toMatchSnapshot();
    });
});