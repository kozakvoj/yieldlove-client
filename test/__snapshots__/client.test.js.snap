// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client should return empty data 1`] = `Object {}`;

exports[`client should return filled data 1`] = `
Object {
  "wemgehoert.de": Row {
    "CPM (Euro)": "1,20",
    "CTR (%)": "0,00",
    "Clicks": "296",
    "Date": "2018-10-01 - 2018-10-01",
    "Impressions": "96744",
    "Placement": "SUM",
    "Revenue (Euro)": "116,51",
  },
}
`;
